agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -59,6 +65,10 @@ def ensure_desired_state
         else
           warn "unknown state #{service_pod.desired_state} for #{service_pod.name}"
         end
+
+        if service_pod.running? && service_container
```

Comment:
These conditions only apply to healthcheck, so maybe it would make sense to do them in `ensure_health_check`?